### PR TITLE
Free maintenance

### PR DESCRIPTION
**Pull request by ghost**
Hi @rodaine. I found your PoC very informative, so I took upon myself to do a bit of maintenance work on it.

* Updated Go toolchain to 1.19 in Dockerfile and turned it into a smaller "from scratch" Dockerfile following these guidelines: https://github.com/jeremyhuiskamp/golang-docker-scratch
* Updated Go protobuf and gRPC autogenerated code
* Updated dependencies
* Removed usages of the deprecated `github.com/golang/protobuf` module in favor of `google.golang.org/protobuf`

### CODE REVIEW

File: protos/chat.proto [hunk 1]
```diff
@@ -4,7 +4,7 @@ package chat;
 
 import "google/protobuf/timestamp.proto";
 
-option go_package="github.com/rodaine/grpc-chat/protos";
+option go_package="./protos";
```
Comment (rodaine) [inline protos/chat.proto]: So we want to keep this the same as it drives what dependent code would use when generating import statements in go. You should be able to modify the `go_out` and `go-grpc_out` options passed to protoc to use the same directory path with [`paths=source_relative`](https://developers.google.com/protocol-buffers/docs/reference/go-generated#invocation)

**Review by rodaine — COMMENTED** [review]
Hey! Thanks so much for this patch. Let me know if you have any questions about the `source_relative` bits I mentioned below. 😁

**Comment by ghost** [comment]
Sorry man, I didn't see the notification.

I've restored the package name and found the correct incantation for protoc (thanks for the pointers). It's in the last commit.

Cheers.

**Comment by rodaine** [comment]
Thanks again!

**Pull request merged**